### PR TITLE
lecture4

### PR DESCRIPTION
**Pull request by LeraShagoyko**

### CODE REVIEW

File: lecture4/src/main/java/Helper.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    public Helper() {
+        this.client = HttpClients.createDefault();
+        this.userName = "Lera" + System.currentTimeMillis() + "@gmail.com";
+        this.password = "password" + System.currentTimeMillis();
+        this.content = "content" + System.currentTimeMillis();
+        this.mapper = new ObjectMapper();
+    }
+
+    public void registrate() {
+        User newUser = new User(this.userName, this.password);
```
Comment (acherkas) [inline lecture4/src/main/java/Helper.java]: where do you use newUser also?
you should work with instance User
return User

File: lecture4/src/main/java/Helper.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class Helper {
+    private CloseableHttpClient client;
+    private String userName;
```
Comment (acherkas) [inline lecture4/src/main/java/Helper.java]: move to User instance

File: lecture4/src/main/java/Helper.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    private String password;
```
Comment (acherkas) [inline lecture4/src/main/java/Helper.java]: move to User instance

File: lecture4/src/main/java/Helper.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    private String content;
+    private String secretKey;
```
Comment (acherkas) [inline lecture4/src/main/java/Helper.java]: move to User instance

File: lecture4/src/main/java/Helper.java (hunk 1 above)
Comment (acherkas) [inline lecture4/src/main/java/Helper.java]: use in arguments (password" + System.currentTimeMillis(); etc)

File: lecture4/src/main/java/NoteContent.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+public class NoteContent {
```
Comment (acherkas) [inline lecture4/src/main/java/NoteContent.java:1]: Not necessary  class you can use just string

File: lecture4/src/main/java/Helper.java [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    private ObjectMapper mapper;
+
+
+    public Helper() {
```
Comment (acherkas) [inline lecture4/src/main/java/Helper.java]: please create LoginHelper and move register() and login() methods there

File: lecture4/src/main/java/Helper.java [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class Helper {
```
Comment (acherkas) [inline lecture4/src/main/java/Helper.java]: rename to NoteHelper or NoteService

File: lecture4/src/main/java/Helper.java (hunk 1 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/Helper.java]: done

File: lecture4/src/main/java/Helper.java (hunk 2 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/Helper.java]: done

File: lecture4/src/main/java/Helper.java (hunk 3 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/Helper.java]: done

File: lecture4/src/main/java/Helper.java (hunk 4 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/Helper.java]: done

File: lecture4/src/main/java/Helper.java (hunk 1 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/Helper.java]: done

File: lecture4/src/main/java/NoteContent.java (hunk 5 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/NoteContent.java:1]: We need to use json
json

File: lecture4/src/main/java/Helper.java (hunk 6 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/Helper.java]: done

File: lecture4/src/main/java/Helper.java (hunk 7 above)
Comment (LeraShagoyko) [reply lecture4/src/main/java/Helper.java]: done

**Pull request merged**